mitodl/open-discussions
# Fix profile celery task race condition

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #2556 

#### What's this PR do?
This moves the triggering of a few celery tasks to a `transaction.on_commit` callback so that we're guaranteed that the celery worker sees the new/updated profile record.

#### How should this be manually tested?
Exercise the app where the profile is getting modified and make sure the task is still firing and there are no errors. A few key areas:

- Register a new user via social auth (email signup is sufficient)
- Register a new user in micromasters and verify they successfully get created in discussions
- Update the user's profile
